foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/database/Oracle8Platform.java

```diff
@@ -14,12 +14,274 @@
 
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Hashtable;
+
+import org.eclipse.persistence.internal.databaseaccess.DatabaseCall;
+import org.eclipse.persistence.internal.databaseaccess.FieldTypeDefinition;
+import org.eclipse.persistence.internal.databaseaccess.Platform;
+import org.eclipse.persistence.internal.databaseaccess.SimpleAppendCallCustomParameter;
+import org.eclipse.persistence.internal.helper.ClassConstants;
+import org.eclipse.persistence.internal.helper.DatabaseField;
+import org.eclipse.persistence.internal.security.PrivilegedAccessHelper;
+import org.eclipse.persistence.internal.security.PrivilegedGetContextClassLoader;
+import org.eclipse.persistence.internal.sessions.AbstractSession;
+import org.eclipse.persistence.logging.AbstractSessionLog;
+import org.eclipse.persistence.logging.SessionLog;
+import org.eclipse.persistence.platform.database.OraclePlatform;
+import org.eclipse.persistence.queries.Call;
+
 /**
  * <p><b>Purpose:</b>
  * Provides Oracle version specific behavior when 
  * org.eclipse.persistence.oracle bundle is not available.
  */
 public class Oracle8Platform extends OraclePlatform {
+    /**
+     * Locator is required for Oracle thin driver to write LOB value exceeds the
+     * limits
+     */
+    protected boolean usesLocatorForLOBWrite = true;
+
+    /** The LOB value limits when the Locator is required for the writing */
+    protected int lobValueLimits = 0;
+
+    /**
+     * INTERNAL:
+     */
+    @Override
+    protected Hashtable buildFieldTypes() {
+        Hashtable fieldTypeMapping = super.buildFieldTypes();
+
+        fieldTypeMapping.put(Byte[].class, new FieldTypeDefinition("BLOB", false));
+        fieldTypeMapping.put(Character[].class, new FieldTypeDefinition("CLOB", false));
+
+        return fieldTypeMapping;
+    }
+
+    /**
+     * INTERNAL: Allow for conversion from the Oralce type to the Java type.
+     */
+    @Override
+    public void copyInto(Platform platform) {
+        super.copyInto(platform);
+        if (!(platform instanceof Oracle8Platform)) {
+            return;
+        }
+        Oracle8Platform oracle8Platform = (Oracle8Platform) platform;
+        oracle8Platform.setShouldUseLocatorForLOBWrite(shouldUseLocatorForLOBWrite());
+        oracle8Platform.setLobValueLimits(getLobValueLimits());
+    }
+
+    /**
+     * INTERNAL Used by SQLCall.appendModify(..) If the field should be passed
+     * to customModifyInDatabaseCall, retun true, otherwise false. Methods
+     * shouldCustomModifyInDatabaseCall and customModifyInDatabaseCall should be
+     * kept in sync: shouldCustomModifyInDatabaseCall should return true if and
+     * only if the field is handled by customModifyInDatabaseCall.
+     */
+    @Override
+    public boolean shouldUseCustomModifyForCall(DatabaseField field) {
```

Comment:
Overridden by org.eclipse.persistence.platform.database.oracle.Oracle9Platform, but may fall into the same NClob question.